src/fmt.rs

```diff
@@ -55,3 +60,118 @@ impl ErrorImpl<()> {
         Ok(())
     }
 }
+
+struct Indented<'a, D> {
+    inner: &'a mut D,
+    ind: Option<usize>,
+    started: bool,
+}
+
+impl<'a, D> Indented<'a, D> {
+    fn numbered(inner: &'a mut D, ind: usize) -> Self {
+        Self {
+            inner,
+            ind: Some(ind),
+            started: false,
+        }
+    }
+
+    fn new(inner: &'a mut D) -> Self {
+        Self {
+            inner,
+            ind: None,
+            started: false,
+        }
+    }
+}
+
+impl<T> fmt::Write for Indented<'_, T>
+where
+    T: fmt::Write,
+{
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        for (ind, mut line) in s.split('\n').enumerate() {
```

Comment:
I have renamed this `ind` to `i` and renamed `self.ind` to `self.number`. I found it confusing to have two `ind` variables referring to different things, and also at the same time being ambiguous between "index" vs "indent" in the context of an indented writer. For example one could imagine an indented writer where `self.ind` specifies the indent amount like 4 or indent string like <code>"&nbsp;&nbsp;&nbsp;&nbsp;"</code>.